widget/container.go

```diff
@@ -216,14 +216,24 @@ func (c *Container) PreferredSize() (int, int) {
 		}
 	}
 
-	// If the set MinHeight or MinWidth are greater than calculated, use that
+	// If w & h are less than the Min values, use the Min values
 	if c.widget != nil && h < c.widget.MinHeight {
 		h = c.widget.MinHeight
 	}
 	if c.widget != nil && w < c.widget.MinWidth {
 		w = c.widget.MinWidth
 	}
 
+	// If w & h are greater than the Max values (and they are set), use the Max values
+	if c.widget != nil && (c.widget.MaxHeight != 0 || c.widget.MaxWidth != 0) {
```

Comment:
So there is a small issue here. If a user can't set just a MaxHeight or MaxWidth, they'd have to set both. I think this if statement should be split up for MaxHeight and MaxWidth